apps/api/v2/src/ee/schedules/schedules.repository.ts

```diff
@@ -64,6 +96,81 @@ export class SchedulesRepository {
     return schedule;
   }
 
+  async updateSchedule(
+    userId: number,
+    scheduleId: number,
+    schedule: Partial<Omit<InputScheduleTransformed, "isDefault">>
+  ) {
+    const { availability, overrides } = schedule;
+
+    const updateScheduleData: Prisma.ScheduleUpdateInput = {
+      name: schedule.name,
+      timeZone: schedule.timeZone,
+    };
+
+    const availabilitiesAndOverrides: Prisma.AvailabilityCreateManyInput[] = [];
+
+    const deleteConditions = [];
+    if (availability) {
+      deleteConditions.push({
+        scheduleId: { equals: scheduleId },
+        date: null,
+      });
+    }
+
+    if (overrides) {
+      deleteConditions.push({
+        scheduleId: { equals: scheduleId },
+        NOT: { date: null },
+      });
+    }
```

Comment:
I think this part with delete conditions can use comments to give some context

Reply:
Done in [refactor: comment schedules repository update logic](https://github.com/calcom/cal.com/pull/14870/commits/9e8606ba3f96678d44ad6f7fcf8392fd222ae72e)